hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/EvmTxProcessor.java

```diff
@@ -112,7 +112,13 @@ protected TransactionProcessingResult execute(
             final BigInteger userOfferedGasPrice,
             final long maxGasAllowanceInTinybars,
             final Account relayer) {
-        final Wei gasCost = Wei.of(Math.multiplyExact(gasLimit, gasPrice));
+
+        Wei gasCost;
+        {
+            final Wei wGasLimit = Wei.of(gasLimit);
+            final Wei wGasPrice = Wei.of(gasPrice);
```

Comment:
Wei has a long multiply, so only one needs to be converted to Wei.

Reply:
fixed.